authors/apps/authentication/views.py

```diff
@@ -143,3 +146,54 @@ def update(self, request, *args, **kwargs):
         serializer.save()
 
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class PasswordResetView(CreateAPIView):
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = EmailSerializer
+
+    def post(self, request):
+        recipient = request.data.get('email', {})
+        if not recipient:
+            return Response({"message": "Email field cannot be blank"}, status=status.HTTP_400_BAD_REQUEST)
+
+        token = jwt.encode({"email": recipient},
+                           settings.SECRET_KEY, algorithm='HS256')
+        
+        is_user_exising = User.objects.filter(email=recipient).exists()
+        if is_user_exising:
+            result = MailSender.send_email_message(recipient, token, request)
```

Comment:
Is this method hit during the testing? Because if so, it will try to actually send an email and we cannot be access outside servers during testing and thus we'll have to mock the sending of the email.

Reply:
let me check it

Reply:
Hi, the `MailSender.send_email_message()` method is hit during testing. When hit, it invokes the `send_mail()` function in `mail.py` which is responsible for actually sending the mail.

During testing, i think sending the mail is mocked(because it doesnt actually happen but a response is obtained), and since the tests are meant to **ascertain the response from the request made**,  the **message responses & status codes** obtained are what i actually tested for.

Thats what i think of the scenario,kindly give more details of how i should go about mocking the mail sending functionality. Thanks